packages/@storefront/flux-capacitor/test/unit/core/observer.ts

```diff
@@ -288,6 +288,84 @@ suite('Observer', ({ expect, spy, stub }) => {
 
+      describe('history()', () => {
+        it('should not emit if state is identical to old state', () => {
+          const state = { a: 'b' };
+
+          observers.data.present.history(<any>state, <any>state, path);
+
+          expect(emit).to.not.be.called;
+        });
+
+        it('should emit URL_UPDATED when urls differ', () => {
+          const oldState = { url: 'www.google.ca' };
+          const newState = { url: 'www.example.ca' };
+
+          observers.data.present.history(oldState, newState, path);
+
+          expect(emit).to.be.calledWith(Events.URL_UPDATED);
+        });
+
+        it('should emit SEARCH_URL_UPDATED when urls differ, shouldFetch is true, and route is search', () => {
+          const oldState = { url: 'www.google.ca', route: utils.Routes.DETAILS, shouldFetch: false };
+          const newState = { url: 'www.example.ca', route: utils.Routes.SEARCH, shouldFetch: true };
+
+          observers.data.present.history(oldState, newState, path);
+
+          expect(emit).to.be.calledWith(Events.SEARCH_URL_UPDATED);
+        });
+
+        it('should emit DETAILS_URL_UPDATED when urls differ, shouldFetch is true, and route is details', () => {
+          const oldState = { url: 'www.google.ca', route: utils.Routes.SEARCH, shouldFetch: false };
+          const newState = { url: 'www.example.ca', route: utils.Routes.DETAILS, shouldFetch: true };
+
+          observers.data.present.history(oldState, newState, path);
+
+          expect(emit).to.be.calledWith(Events.DETAILS_URL_UPDATED);
+        });
+
+        // tslint:disable-next-line max-line-length
+        it('should emit PAST_PURCHASE_URL_UPDATED when urls differ, shouldFetch is true, and route is past purchase', () => {
+          const oldState = { url: 'www.google.ca', route: utils.Routes.SEARCH, shouldFetch: false };
+          const newState = { url: 'www.example.ca', route: utils.Routes.PAST_PURCHASE, shouldFetch: true };
+
+          observers.data.present.history(oldState, newState, path);
+
+          expect(emit).to.be.calledWith(Events.PAST_PURCHASE_URL_UPDATED);
+        });
+
+        it('should emit CUSTOM_URL_UPDATED when urls differ, shouldFetch is true, and route is not known', () => {
+          const oldState = { url: 'www.google.ca', route: utils.Routes.SEARCH, shouldFetch: false };
+          const newState = { url: 'www.example.ca', route: null, shouldFetch: true };
+
+          observers.data.present.history(oldState, newState, path);
+
+          expect(emit).to.be.calledWith(Events.CUSTOM_URL_UPDATED);
+        });
+
+        it('should not emit when urls differ, and shouldFetch is false', () => {
+          const oldState = { url: 'www.google.ca', route: utils.Routes.SEARCH, shouldFetch: false };
+          const newState = { url: 'www.example.ca', route: utils.Routes.DETAILS, shouldFetch: false };
+
+          observers.data.present.history(oldState, newState, path);
+
+          expect(emit).to.be.calledWith(Events.URL_UPDATED);
```

Comment:
Can we assume that `emit` is only called once? If so, we won't need the checks below.